x-pack/plugins/enterprise_search/server/routes/app_search/app_search_request_handler.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import fetch from 'node-fetch';
+import querystring from 'querystring';
+import { Logger } from 'src/core/server';
+import { ConfigType } from '../../index';
+
+interface IAppSearchRequestParams {
+  config: ConfigType;
+  log: Logger;
+  path: string;
+  hasValidData: (body) => boolean;
+}
+
+/**
+ * This function creates a handler, in a standard way, for handling App Search
+ * API requests.
+ *
+ * This handler assumes that it will essentially just proxy the App Search API
+ * request, so the request body and request parameters, and body are simply
+ * passed through.
+ */
+export function createAppSearchRequestHandler({
```

Comment:
This is a super great helper, I love it! :)

Can we move this into `server/lib/request_handler.ts` instead of making it App Search specific (also since it's not really a true "route")? That way WS and generic ES routes can use it as well - what do you think?